apps/07tic-tac-toe/script.js

```diff
@@ -2,4 +2,170 @@
+        board[2][0] === playerTurn && board[2][1] === playerTurn && board[2][2] === playerTurn) {
+        return true;
+    }
+        else {
+            return false;
+        }
+    }
+
+function verticalWin() {
+    if (board[0][0] === playerTurn && board[1][0] === playerTurn && board[2][0] === playerTurn ||
+        board[0][1] === playerTurn && board[1][1] === playerTurn && board[2][1] === playerTurn ||
+        board[0][2] === playerTurn && board[1][2] === playerTurn && board[2][2] === playerTurn) {
+        return true;
+    }
+        else {
+            return false;
+        }
+    }
+
+function diagonalWin() {
+    if (board[0][0] === playerTurn && board[1][1] === playerTurn && board[2][2] === playerTurn ||
+        board[0][2] === playerTurn && board[1][1] === playerTurn && board[2][0] === playerTurn) {
+        return true;
+    }
+        else {
+            return false;
+        }
+    }
+
+    
+    function checkForWin() {
+    if (horizontalWin() || verticalWin() || diagonalWin()) {
+        return true
+        console.log('Player' + playerTurn + 'Won!');
+    } else {
+        return false
+    }
+}
+    
+    function isDraw() {
+    if (board[0][0] === playerTurn && board[0][1] === playerTurn && board[0][2] === playerTurn &&
+        board[1][0] === playerTurn && board[1][1] === playerTurn && board[1][2] === playerTurn &&
+        board[2][0] === playerTurn && board[2][1] === playerTurn && board[2][2] === playerTurn) {
+        return true
+        console.log('Its a Draw');
+    } else {
+        return false
+    }
+}
+    
+    function checkForDraw() {
+
+    var emptySpots = 9;
+
+    for (var i = 0; i < board.length; i++) {
+        for (var j = 0; j < board[i].length; j++) {
+            if (board[i][j] != " ") {
+                emptySpots--;
+            }
+        }
+    }
+
+    if (emptySpots === 0 && !checkForWin()) {
+        console.log("\nCat's Game! Let's Reset...\n");
+        resetBoard();
+    }
+}
+    
+    function ticTacToe(row, column) {
+    if (board[row][column] == " ") {
+        if (row == 0 && column == 0) {
+            board[0][0] = playerTurn;
+        }
+        if (row == 0 && column == 1) {
+            board[0][1] = playerTurn;
+        }
+        if (row == 0 && column == 2) {
+            board[0][2] = playerTurn;
+        }
+
+
+        if (row == 1 && column == 0) {
+            board[1][0] = playerTurn;
+        }
+        if (row == 1 && column == 1) {
+            board[1][1] = playerTurn;
+        }
+        if (row == 1 && column == 2) {
+            board[1][2] = playerTurn;
+        }
+
+        if (row == 2 && column == 0) {
+            board[2][0] = playerTurn;
+        }
+        if (row == 2 && column == 1) {
+            board[2][1] = playerTurn;
+        }
+        if (row == 2 && column == 2) {
+            board[2][2] = playerTurn;
+        }
+    }
+
+    if (checkForWin() == true) {
```

Comment:
It is usually considered bad practice to use true or false in conditionals (ie. someVariable == true, someFunction() ==  false, etc.). A better approach is to utilize the return value of the function:

`
if(checkForWin()){
//do something
}
`